rust/kernel/allocator.rs

```diff
@@ -2,19 +2,20 @@
 
 //! Allocator support.
 
-use core::alloc::{GlobalAlloc, Layout};
-use core::ptr;
+use core::alloc::{AllocError, Allocator, GlobalAlloc, Layout};
+use core::ptr::{self, NonNull};
 
 use crate::bindings;
 use crate::c_types;
 
-struct KernelAllocator;
+#[derive(Copy, Clone)]
+pub struct KernelAllocator<const A: bindings::gfp_t>;
 
-unsafe impl GlobalAlloc for KernelAllocator {
+unsafe impl<const A: bindings::gfp_t> GlobalAlloc for KernelAllocator<A> {
```

Comment:
I think it makes sense to limit the `GlobalAlloc` impl to `KernelAllocator<{ bindings::GFP_KERNEL }>`.

Reply:
Good catch. Did not think of that

Reply:
Fixed